cache/src/main/scala/org/allenai/common/cache/JsonQueryCacheModule.scala

```diff
@@ -4,14 +4,25 @@ import org.allenai.common.guice.ConfigModule
 
 import com.google.inject.Provides
 import com.google.inject.name.Named
-import com.typesafe.config.{ Config, ConfigFactory }
+import com.typesafe.config.Config
 import redis.clients.jedis.{ JedisPool, JedisPoolConfig }
+import spray.json.JsonFormat
+
+/** A guice module for binding a `JsonQueryCache` instance with a parameterized value type. */
+class JsonQueryCacheModule[V](config: Config)(
+  implicit
+  jsonFormat: JsonFormat[V],
+  manifest: Manifest[V]
+) extends ConfigModule(config) {
 
-class JsonQueryCacheModule(config: Config = ConfigFactory.empty) extends ConfigModule(config) {
   override def configName: String = "cache.conf"
 
   override def bindingPrefix: Option[String] = Some("redis")
 
+  override def configureWithConfig(fullConfig: Config): Unit = {
+    bind[JsonFormat[V]].toInstance(jsonFormat)
+  }
+
   @Provides def provideJedisPool(
```

Comment:
Wouldn't that then force all caches bound to an injector to point at the same Redis host?  I thought that wasn't what we wanted.